### PR TITLE
fix: wallet unlock

### PR DESCRIPTION
**Pull request by nuanyang233**
closes #

### CODE REVIEW

File: packages/dashboard/src/locales/en-US.json [hunk 1]
```diff
@@ -325,7 +325,8 @@
     "create_wallet_password_lowercase_tip": "Must contain a lowercase character",
     "create_wallet_password_number_tip": "Must contain a number",
     "create_wallet_password_special_tip": "Must contain a special character",
-    "create_wallet_password_match_tip": "Passwords don't match",
+    "create_wallet_password_match_tip": "The code does not match",
+    "create_wallet_password_length_error": "The password length is incorrect",
```
Comment (guanbinrui) [inline packages/dashboard/src/locales/en-US.json:329]:
```suggestion
    "create_wallet_password_length_error": "The password length is incorrect.",
```

File: packages/dashboard/src/locales/en-US.json [hunk 2]
```diff
@@ -325,7 +325,8 @@
     "create_wallet_password_lowercase_tip": "Must contain a lowercase character",
     "create_wallet_password_number_tip": "Must contain a number",
     "create_wallet_password_special_tip": "Must contain a special character",
-    "create_wallet_password_match_tip": "Passwords don't match",
+    "create_wallet_password_match_tip": "The code does not match",
```
Comment (guanbinrui) [inline packages/dashboard/src/locales/en-US.json:328]:
```suggestion
    "create_wallet_password_match_tip": "The password does not match.",
```

File: packages/maskbook/src/locales/en-US.json [hunk 3]
```diff
@@ -876,6 +876,8 @@
     "popups_wallet_contract_interaction_gas_fee": "Gas fee",
     "popups_wallet_contract_interaction_edit": "Edit",
     "popups_wallet_contract_interaction_total": "Total",
+    "popups_wallet_password_dont_match": "The code does not match",
+    "popups_wallet_password_length_error": "The password length is incorrect",
```
Comment (guanbinrui) [inline packages/maskbook/src/locales/en-US.json:880]:
```suggestion
    "popups_wallet_password_dont_match": "The password does not match.",
    "popups_wallet_password_length_error": "The password length is incorrect.",
```

**Review by guanbinrui — DISMISSED** [review]

File: packages/maskbook/src/extension/background-script/WelcomeServices/generateBackupJSON.ts [hunk 4]
```diff
@@ -65,6 +65,10 @@ export async function generateBackupJSON(opts: Partial<BackupOptions> = {}): Pro
         userGroups: [],
     }
     if (Object.keys(plugins).length) file.plugin = plugins
+
+    console.log('DEBUG: generateBackupJSON')
+    console.log(file)
+
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/background-script/WelcomeServices/generateBackupJSON.ts:71]:
```suggestion
```

File: packages/maskbook/src/extension/options-page/DashboardDialogs/Wallet/DeleteConfirm.tsx [hunk 5]
```diff
@@ -13,7 +13,7 @@ export function DashboardWalletDeleteConfirmDialog(props: WrappedDialogProps<Wal
     const { wallet } = props.ComponentProps!
     const onConfirm = useSnackbarCallback(
         async () => {
-            await WalletRPC.removeWallet(wallet.address)
+            await WalletRPC.deleteWallet(wallet.address)
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardDialogs/Wallet/DeleteConfirm.tsx:16]: `removeWallet`?

File: packages/maskbook/src/extension/popups/pages/Swap/index.tsx [hunk 6]
```diff
@@ -60,7 +61,9 @@ export default function SwapPage() {
     const { value: pendingTransactions = [] } = useRecentTransactions(TransactionStatusType.NOT_DEPEND)
     const wallet = useWallet()
     const openPopupsWindow = useCallback(() => {
-        Services.Helper.openPopupsWindow('/wallets')
+        Services.Helper.openPopupsWindow('/wallet/select', {
+            chainId: chainId,
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Swap/index.tsx:65]:
```suggestion
            chainId,
```

File: packages/maskbook/src/extension/popups/pages/Swap/index.tsx [hunk 7]
```diff
@@ -60,7 +61,9 @@ export default function SwapPage() {
     const { value: pendingTransactions = [] } = useRecentTransactions(TransactionStatusType.NOT_DEPEND)
     const wallet = useWallet()
     const openPopupsWindow = useCallback(() => {
-        Services.Helper.openPopupsWindow('/wallets')
+        Services.Helper.openPopupsWindow('/wallet/select', {
+            chainId,
+        })
     }, [])
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Swap/index.tsx:67]:
```suggestion
    }, [chainId])
```

File: packages/maskbook/src/extension/popups/pages/Wallet/Unlock/index.tsx [hunk 8]
```diff
@@ -52,9 +52,9 @@ const Unlock = memo(() => {
     const [password, setPassword] = useState('')
 
     const history = useHistory()
-    const [unlocked, handleUnlock] = useAsyncFn(async () => {
-        await WalletRPC.unlockWallet(password)
-        history.replace(PopupRoutes.Wallet)
+    const [{ value: hasError, loading }, handleUnlock] = useAsyncFn(async () => {
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/popups/pages/Wallet/Unlock/index.tsx:55]:
```suggestion
    const [{ value: unlocked, error: hasError, loading }, handleUnlock] = useAsyncFn(async () => {
```

File: packages/maskbook/src/locales/en-US.json [hunk 9]
```diff
@@ -834,6 +834,9 @@
     "plugin_unlockprotocol_server_error": "Some Server error occured, Please try again later.",
     "popups_welcome": "Welcome",
     "popups_wallet_token": "Token",
+    "popups_wallet_set_payment_password": "Set the payment password",
+    "popups_wallet_payment_password": "Payment Password",
+    "popups_wallet_payment_password_tip": "Payment Password must be a combination of 2 categories out of numbers, letters and special characters with a length of 8-20 characters.",
```
Comment (guanbinrui) [inline packages/maskbook/src/locales/en-US.json:839]:
```suggestion
    "popups_wallet_payment_password_tip": "The Payment password must be a combination of 2 categories out of numbers, letters, and special characters with a length of 8-20 characters.",
```

File: packages/maskbook/src/locales/en-US.json (hunk 3 above)
Comment (guanbinrui) [reply packages/maskbook/src/locales/en-US.json:883]: Please add `.` at the end of a sentence.

File: packages/maskbook/src/plugins/Wallet/services/wallet/database/wallet.ts [hunk 10]
```diff
@@ -140,12 +136,7 @@ export async function updateWallet(
     WalletMessages.events.walletsUpdated.sendToAll(undefined)
 }
 
-export async function removeWallet(address: string) {
-    const wallet = await getWalletRequired(address)
-
-    // delete a wallet with mnemonic is not allowed
-    if (wallet.derivationPath) throw new Error('Illegal operation.')
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/wallet/database/wallet.ts:147]: Why remove this?

File: packages/maskbook/src/plugins/Wallet/services/wallet/index.ts [hunk 11]
```diff
@@ -9,6 +9,7 @@ import * as sdk from './maskwallet'
 import * as database from './database'
 import * as password from './password'
 import type { TransactionConfig } from 'web3-core'
+import { deleteWallet } from './database'
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/wallet/index.ts:12]: Please use `database.deleteWallet`.

File: packages/maskbook/src/plugins/Wallet/services/wallet/password.ts [hunk 12]
```diff
@@ -52,7 +54,7 @@ export function validatePassword(unverifiedPassword: string) {
 }
 
 export function validatePasswordRequired(unverifiedPassword: string) {
-    if (!validatePassword(unverifiedPassword)) throw new Error('An invalid password.')
+    if (!validatePassword(unverifiedPassword)) throw new Error('Incorrect payment password')
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/wallet/password.ts:57]: This error means the password does not adhere to the basic validation rules. It hasn't check if the password is correct or not yet.

File: packages/maskbook/src/plugins/Wallet/services/chain.ts [hunk 13]
```diff
@@ -31,10 +33,14 @@ export async function updateChainState() {
...
     // update chain state
     try {
-        ;[currentBlockNumberSettings.value, currentBalanceSettings.value] = await Promise.all([
-            getBlockNumber(),
-            currentAccountSettings.value ? getBalance(currentAccountSettings.value) : currentBalanceSettings.value,
-        ])
+        ;[currentBlockNumberSettings.value, currentBalanceSettings.value, currentMaskWalletBalanceSettings.value] =
+            await Promise.all([
+                getBlockNumber(),
+                currentAccountSettings.value ? getBalance(currentAccountSettings.value) : currentBalanceSettings.value,
```
Comment (guanbinrui) [inline packages/maskbook/src/plugins/Wallet/services/chain.ts:39]: Please reduce request when `currentAccountSettings` is the same as the `currentAccountMaskWalletSettings`.

File: packages/maskbook/src/utils/type-transform/BackupFormat/JSON/version-2.ts [hunk 14]
```diff
@@ -75,8 +70,6 @@ export interface BackupJSONFileVersion2 {
         passphrase?: string
         publicKey?: EC_Public_JsonWebKey
         privateKey?: EC_Private_JsonWebKey
-        derivationPath?: string
-        storedKeyInfo?: api.IStoredKeyInfo
```
Comment (Jack-Works) [inline packages/maskbook/src/utils/type-transform/BackupFormat/JSON/version-2.ts:79]:
```ts
    /** @deprecated No longer used */
    storedKeyInfo?: unknown
```

File: packages/maskbook/src/web3/context.ts [hunk 15]
```diff
@@ -70,7 +59,8 @@ function createWeb3Context(disablePopup = false): Web3ProviderType {
 }
 
 export const Web3Context = createWeb3Context()
-export const Web3ContextWithoutConfirm = createWeb3Context(true)
+export const PopupWeb3Context = createWeb3Context(true, true)
+export const SwapWeb3Context = createWeb3Context(false, true)
```
Comment (guanbinrui) [inline packages/maskbook/src/web3/context.ts:63]: 👏

Comment (guanbinrui) [reply packages/maskbook/src/web3/context.ts:63]: MaskApplicationWeb3Context

File: packages/maskbook/src/web3/context.ts [hunk 16]
```diff
@@ -70,7 +59,8 @@ function createWeb3Context(disablePopup = false): Web3ProviderType {
 }
 
 export const Web3Context = createWeb3Context()
-export const Web3ContextWithoutConfirm = createWeb3Context(true)
+export const PopupWeb3Context = createWeb3Context(true, true)
```
Comment (guanbinrui) [inline packages/maskbook/src/web3/context.ts:62]: `MaskWeb3Context`

File: packages/maskbook/src/utils/type-transform/BackupFormat/JSON/version-2.ts (hunk 14 above)
Comment (guanbinrui) [reply packages/maskbook/src/utils/type-transform/BackupFormat/JSON/version-2.ts:79]: I just added in Mask 2.0.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**